File: 🟡 Medium/921 - Minimum Add to Make Parentheses Valid/_921MinimumAddtoMakeParenthesesValid.java
class Solution {
    public int minAddToMakeValid(String s) {
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            if (!stack.isEmpty() && stack.peek() == '(' && s.charAt(i) == ')') {
                stack.pop();
            } else {
                stack.push(s.charAt(i));
            }
        }

        return stack.size();
    }
}

File: 🟡 Medium/921 - Minimum Add to Make Parentheses Valid/_921MinimumAddtoMakeParenthesesValid.js
/**
 * @param {string} s
 * @return {number}
 */
var minAddToMakeValid = function (s) {
    let stack = [];

    for (let i = 0; i < s.length; i++) {
        if (stack.length != 0 && stack.slice(-1) == '(' && s[i] == ')') {
            stack.pop();
        } else {
            stack.push(s[i]);
        }
    }

    return stack.length;
};